packages/core/state/interaction/logics.ts

```diff
@@ -582,17 +582,78 @@ const setIsSmallScreen = createLogic({
-            console.log("Cache statuses:", cacheStatuses);
+            const fileDetails = (action as MoveFilesAction).payload.fileDetails;
+
+            // Map file IDs to file names for easy lookup
+            const fileIdToNameMap = Object.fromEntries(
+                fileDetails.map((file) => [file.id, file.name])
+            );
+
+            // Extract file IDs
+            const fileIds = fileDetails.map((file) => file.id);
+
+            const response = await httpFileService.cacheFiles(fileIds, username);
+            const cacheStatuses = response.cacheFileStatuses;
+
+            // Check if the response is empty.
+            if (!cacheStatuses || Object.keys(cacheStatuses).length === 0) {
+                dispatch(
+                    interaction.actions.processError(
+                        "moveFilesNoFilesProcessed",
+                        "No files were processed. Please check the request or try again."
+                    )
+                );
+                return;
+            }
+
+            const successfulFiles: string[] = [];
+            const failedFiles: string[] = [];
+
+            Object.entries(cacheStatuses).forEach(([fileId, status]) => {
+                if (
+                    status === "DOWNLOAD_COMPLETE" ||
+                    status === "DOWNLOAD_IN_PROGRESS" ||
+                    status === "DOWNLOAD_STARTED"
+                ) {
+                    successfulFiles.push(fileId);
+                } else {
+                    failedFiles.push(fileId);
+                }
+            });
+
+            // Dispatch net success message. (Assuming some files were successful)
+            if (successfulFiles.length > 0) {
+                dispatch(
+                    interaction.actions.processSuccess(
+                        "moveFilesSuccess",
+                        `${successfulFiles.length} out of ${
+                            Object.keys(cacheStatuses).length
+                        } files were successfully queued for download to NAS (Vast) from cloud. 
+                        Files will be available in the NAS after downloads finish asynchronously`
+                    )
+                );
+            }
+
+            // Dispatch individual errors for each failed file.
+            failedFiles.forEach((fileId) => {
+                const fileName = fileIdToNameMap[fileId] || "Unknown File";
+                dispatch(
+                    interaction.actions.processError(
+                        `moveFileFailure_${fileId}`,
+                        `File "${fileName}" failed to cache. Status: ${cacheStatuses[fileId]}`
+                    )
+                );
+            });
         } catch (err) {
-            throw new Error(`Error encountered while moving files: ${err}`);
+            // Service call itself fails
+            dispatch(
+                interaction.actions.processError("moveFilesFailure", `Failed to cache files.`)
```

Comment:
Might be good just to toss the error from the err.message here like
```suggestion
                interaction.actions.processError("moveFilesFailure", `Failed to cache files, details: ${(err as Error).message}.`)
```